packages/relay/src/lib/config/hbarSpendingPlanConfigService.ts

```diff
@@ -108,16 +109,19 @@ export class HbarSpendingPlanConfigService {
    * @private
    */
   private loadSpendingPlansConfig(): SpendingPlanConfig[] {
-    const configPath = findConfig(this.SPENDING_PLANS_CONFIG_FILE);
-    if (!configPath || !fs.existsSync(configPath)) {
-      this.logger.trace(`Configuration file not found at path "${configPath ?? this.SPENDING_PLANS_CONFIG_FILE}"`);
-      return [];
+    const configPath = findConfig(this.SPENDING_PLANS_CONFIG);
+    if (!this.SPENDING_PLANS_CONFIG) {
+      throw new Error('SPENDING_PLANS_CONFIG is undefined');
     }
     try {
-      const rawData = fs.readFileSync(configPath, 'utf-8');
-      return JSON.parse(rawData) as SpendingPlanConfig[];
+      // Try to parse the value as a file path
+      if (configPath && fs.existsSync(configPath)) {
```

Comment:
If the above check is correct you shouldn't need to recheck the config value
```suggestion
      if (fs.existsSync(configPath)) {
```